### PR TITLE
Salesforce e2e Tests

### PR DESCRIPTION
**Pull request by Vipinofficial11**

### CODE REVIEW

File: src/e2e-test/features/salesforcebatchsource/RunTime.feature [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    And Close the pipeline logs
+    And Close the preview
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Wait till pipeline is in running state
+    And Open and capture logs
+    And Verify the pipeline status is "Succeeded"
+    And Close the pipeline logs
```
Comment (Aryan-Verma) [inline src/e2e-test/features/salesforcebatchsource/RunTime.feature:55]: Can we verify here if the data transferred from the salesforce side is same as the data received in the BQ sink? Please do so for the below scenarios too

File: src/e2e-test/features/salesforcebatchsource/RunTimeWithMacros.feature [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    And Enter runtime argument value from environment variable "admin.security.token" for key "SecurityToken"
+    And Enter runtime argument value from environment variable "admin.consumer.key" for key "ConsumerKey"
+    And Enter runtime argument value from environment variable "admin.consumer.secret" for key "ConsumerSecret"
+    And Enter runtime argument value "login.url" for key "LoginUrl"
+    And Enter runtime argument value "Salesforce.sobjectName" for key "SObjectName"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
```
Comment (Aryan-Verma) [inline src/e2e-test/features/salesforcebatchsource/RunTimeWithMacros.feature:77]: Same data verification comment here for all the scenarios

File: src/e2e-test/features/salesforcesink/RunTime.feature [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
```
Comment (Aryan-Verma) [inline src/e2e-test/features/salesforcesink/RunTime.feature:55]: Same data verification comment here for all scenarios

File: src/e2e-test/features/salesforcebatchsource/DesignTimeWithMacro.feature [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Copyright © 2022 Cask Data, Inc.
```
Comment (Aryan-Verma) [inline src/e2e-test/features/salesforcebatchsource/DesignTimeWithMacro.feature]: nit: 2023
please correct it in every new file

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    }
+  }
+
+  public static void deletePushTopic(String pushTopicName) {
+    try {
+      PartnerConnection partnerConnection = new PartnerConnection(
+        Authenticator.createConnectorConfig(new AuthenticatorCredentials(
+          System.getenv("SALESFORCE_USERNAME"), System.getenv("SALESFORCE_PASSWORD") +
+          System.getenv("SALESFORCE_SECURITY_TOKEN"),
+          System.getenv("SALESFORCE_CONSUMER_KEY"), System.getenv("SALESFORCE_CONSUMER_SECRET"),
```
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: These are all defined above. Can use them

File: src/e2e-test/features/salesforcebatchsource/DesignTimeWithMacro.feature (hunk 4 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/salesforcebatchsource/DesignTimeWithMacro.feature]: updated.[335f1356833ab81ae04213039544120077960c75]

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java (hunk 5 above)
Comment (Vipinofficial11) [reply src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: updated(https://github.com/data-integrations/salesforce/pull/198/commits/335f1356833ab81ae04213039544120077960c75#diff-709912cd5423db1241e3de0b13bb67637b064123b6677f303ce0f82db76c78fbR172)

File: src/e2e-test/features/salesforcebatchsource/RunTime.feature (hunk 1 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/salesforcebatchsource/RunTime.feature:55]: The team is working on the data validation for all the 4 plugins, I believe they will be raising a separate PR for that. Should we add that change in this PR as well or keep a separate one ?

**Review by Aryan-Verma — APPROVED** [review]

**Comment by Vipinofficial11** [comment]
Merging the PR as it has one approval and there are no further review comments.

**Pull request merged**